packages/react-router-dom/server.tsx

```diff
@@ -115,7 +114,14 @@ export function unstable_DataStaticRouter({
           <DataRouterStateContext.Provider
             value={dataRouterContext.router.state}
           >
-            <DataRouter />
+            <Router
+              basename={dataRouterContext.basename}
+              location={dataRouterContext.router.state.location}
+              navigationType={dataRouterContext.router.state.historyAction}
+              navigator={dataRouterContext.navigator}
+            >
+              <Routes />
+            </Router>
```

Comment:
We'll want to rename `DataStaticRouter` to `StaticRouterProvider` since the reciprocal version is named `RouterProvider` ... cant' remember all the other SSR APIs off the top of my head but maybe even be able to just pass a `staticRouter` to `<RouterProvider>` and not even need this component anymore?

Reply:
yeah we still need it for a few SSR-specific things:
* SSR error boundaries
* automatic hydration
* stateless navigator

But we can align the APIs closer to client side via `createStaticRouter`/`StaticRouteProvider` - did that in [fd7f77b](https://github.com/remix-run/react-router/pull/9227/commits/fd7f77bf204fedcfa815d0ee2c9da71559fb767c) 👍